### PR TITLE
Driven Adapter KMS

### PR DESCRIPTION
**Pull request by santitigaga**
## Description

#119  Added KMS driven adapter for reactive and imperative projects.
## Category
- [x] Feature
- [ ] Fix

## Checklist
- [x] The pull request is complete according to the [guide of contributing](https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing)
- [x] Automated tests are written
- [x] The documentation is up-to-date
- [x] the version of the build.gradle, readme.md and gradle.properties was increased
- [x] The pull request has a descriptive title that describes what has changed, and provides enough context for the changelog

### CODE REVIEW

File: src/main/resources/driven-adapter/kms-reactive/kms-adapter.java.mustache [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Base64Utils;
+
+import lombok.RequiredArgsConstructor;
+import reactor.core.publisher.Mono;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.kms.KmsAsyncClient;
+import software.amazon.awssdk.services.kms.model.*;
```
Comment (dericop) [inline src/main/resources/driven-adapter/kms-reactive/kms-adapter.java.mustache]: Do you really  need .* import? This way is usually reported in sonar.

File: src/main/resources/driven-adapter/kms-reactive/kms-connection-properties.java.mustache [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.springframework.context.annotation.Configuration;
+
+import lombok.Data;
+
+{{#lombok}}
+@Data
+{{/lombok}}
+@Configuration
+@ConfigurationProperties(prefix = "adapters.aws.kms")
```
Comment (dericop) [inline src/main/resources/driven-adapter/kms-reactive/kms-connection-properties.java.mustache:12]: did you add the properties in the yaml?

File: src/main/resources/driven-adapter/kms-reactive/build.gradle.mustache [hunk 3]
```diff
@@ -0,0 +1,5 @@
+dependencies {
+    implementation project(':model')
+    compile 'software.amazon.awssdk:kms:2.14.13'
+    compile 'org.springframework:spring-context'
+}
```
Comment (dericop) [inline src/main/resources/driven-adapter/kms-reactive/build.gradle.mustache]: You need to add a newline at the end of mustache files.

**Review by dericop — CHANGES_REQUESTED** [review]

File: src/main/resources/driven-adapter/kms-reactive/kms-adapter.java.mustache (hunk 1 above)
Comment (santitigaga) [reply src/main/resources/driven-adapter/kms-reactive/kms-adapter.java.mustache]: I Agree, I should  updated this

File: src/main/resources/driven-adapter/kms-reactive/kms-connection-properties.java.mustache (hunk 2 above)
Comment (santitigaga) [reply src/main/resources/driven-adapter/kms-reactive/kms-connection-properties.java.mustache:12]: yes, this properties are in upper layer, specifically GenerateDrivenAdapterTask class.

**Review by dericop — APPROVED** [review]

**Pull request merged**